### ISSUE TITLE
how to get results of superglue task Ax-b and Ax-g?

### ISSUE THREAD

**Issue by YoungTimmy** [nyu-mll/jiant#1255]
hi, I am trying to evaluate BERT-base on superglue benchmark with jiant. I followed the example of MNLI diagnostics to modify the codes. However, there is no score printed out, just a list of test_preds. how can I get a result score instead of a list?

thanks a lot :)

**Comment by YoungTimmy** [comment]
and when run WSC task, there is a problem……

![image](https://user-images.githubusercontent.com/39907234/103474646-e9798380-4de0-11eb-8d4c-47cbacc7f5a6.png)

**Comment by jeswan** [comment]
Please see [this guide](https://github.com/nyu-mll/jiant/blob/master/guides/benchmarks/superglue.md) to format the test_preds file for submission to the [SuperGLUE benchmark](https://super.gluebenchmark.com). To get a score on the test sets, you must submit the files to the benchmark website. What score are you looking for in particular?

Thanks for reporting the WSC issue. Will look at this now.

**Comment by jeswan** [comment]
I was unable to reproduce the WSC tokenization error using the steps in https://colab.research.google.com/github/nyu-mll/jiant/blob/master/examples/notebooks/simple_api_fine_tuning.ipynb (uses Huggingface's Datasets instead of the raw task data). Please open up a new issue specifying the BERT model and steps to reproduce the WSC issue if you are still having problems.

**Comment by YoungTimmy** [comment]
> I was unable to reproduce the WSC tokenization error using the steps in https://colab.research.google.com/github/nyu-mll/jiant/blob/master/examples/notebooks/simple_api_fine_tuning.ipynb (uses Huggingface's Datasets instead of the raw task data). Please open up a new issue specifying the BERT model and steps to reproduce the WSC issue if you are still having problems.

thanks for your reply. As to the wsc problem, I installed the new jiant-2.1.3 and the proplem did not show up. I think it is maybe a problem of jiant-2.1.2.

And my previous question is that, I don't know if I can get a metric result of Ax-b and Ax-g task like other tasks such as RTE or Multirc. I followed the example of _**jiant_Basic_Example.ipynb**_ and can get a **_val_metrics.json_** of other 8 tasks. But there is no output of Ax-b and Ax-g task, so how do I know the model's performance on such 2 tasks?  By the way, is the result in **_val_metrics.json_** a **val_acc** or **test_acc** ?

If I want to evaluate my own model on SuperGlue Benchmark with jiant instead of models in [transformers](https://github.com/huggingface/transformers). What should I do?

**Comment by jeswan** [comment]
Ax-b and Ax-g are _diagnostic_ tasks. Please see https://super.gluebenchmark.com/tasks for more information. These tasks only have test sets. `val_metrics.json` contains the score for the validation dataset specified in the task config (generally the validation dataset unless explicitly stated otherwise). 

We are currently in a sprint to make it easier to add a model (this requires refactoring). What architecture is your model? Are you able to upload your model to https://huggingface.co/models?

**Comment by jeswan** [comment]
If you have additional questions unrelated to Ax-b and Ax-g, please open a new issue to help other users navigate Issues.

**Comment by supersteph** [comment]
I'm trying to run AX-g task. However, I don't see how you use the run script such that it doesn't train, I tried adding the write-test-preds flag. What am I supposed to do to run the model?

**Comment by jeswan** [comment]
AX-g is a diagnostic task (see the SuperGLUE info above). For this task, use a pre-trained model (that has already been trained) and only run the test `jiant` with the test set.

**Comment by jeswan** [comment]
I'll post a small example showing that the runscript does not need to train the model

**Comment by jeswan** [comment]
@supersteph To ensure the model does not train, you should simply be able to remove the `--write_test_preds` when running the `main/runscript.py`. Let me know if that makes sense

**Comment by supersteph** [comment]
Would you be able to link a place where python main/runscript.py is used. I have been using simple/runscript.py

**Comment by jeswan** [comment]
Example notebooks (including the main API) can be found here: https://github.com/nyu-mll/jiant/tree/master/examples

**Comment by jeswan** [comment]
Closing due to inactivity.
